### PR TITLE
Create DropTableStatementFederateRefresher for issues#10697

### LINKED ISSUE #10697: Create `DropTableStatementFederateRefresher`
Hi community,

This issue needs more effort than good-first-issue. Hope you can enjoy it. :)

### Target 
- Create a class named `DropTableStatementFederateRefresher` which is to renew `FederateSchemaMetadata` when users execute `drop table xxx` SQL. (It is simple to figure out that once the table structure changes, the metadata needs to be refreshed accordingly)
- Add a unit test for your change

### How
- Create a class named `DropTableStatementFederateRefresher` implementing `FederateRefresher<DropTableStatement>`
Here are some tips,
1. Please refer to `CreateTableStatementFederateRefresher` and `DropTableStatementSchemaRefresher`
2. Add a `remove` function for `FederateSchemaMetadata` like `renew()` of it.
- Add this one to `MetadataRefresherFactory`
- Add a unit test similar to `DropTableStatementSchemaRefresherTest` and #10695

### PR DESCRIPTION
**Pull request by xiaoshen11**
Fixes #10697.

Changes proposed in this pull request:
- Create DropTableStatementFederateRefresher
- add DropTableStatementFederateRefresherTest
-

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-optimize/src/test/java/org/apache/shardingsphere/infra/optimize/core/metadata/refresher/DropTableStatementFederateRefresherTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import org.junit.Test;
+
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.Map;
+
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.mock;
+
+public class DropTableStatementFederateRefresherTest {
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-optimize/src/test/java/org/apache/shardingsphere/infra/optimize/core/metadata/refresher/DropTableStatementFederateRefresherTest.java:28]: Hi it is suggested to be final class.

File: shardingsphere-infra/shardingsphere-infra-optimize/src/main/java/org/apache/shardingsphere/infra/optimize/core/metadata/refresher/type/DropTableStatementFederateRefresher.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * ShardingSphere Federate refresher for drop table statement.
+ */
+public final class DropTableStatementFederateRefresher implements FederateRefresher<DropTableStatement> {
+
+    @Override
+    public void refresh(final FederateSchemaMetadata schema, final Collection<String> routeDataSourceNames, final DropTableStatement sqlStatement, final SchemaBuilderMaterials materials)
+            throws SQLException {
+        sqlStatement.getTables().forEach(each -> schema.remove(each.getTableName().getIdentifier().getValue()));
+        for (SimpleTableSegment each : sqlStatement.getTables()) {
+            ShardingSphereEventBus.getInstance().post(new DropTableEvent(each.getTableName().getIdentifier().getValue()));
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-optimize/src/main/java/org/apache/shardingsphere/infra/optimize/core/metadata/refresher/type/DropTableStatementFederateRefresher.java:41]: Hi it is unnecessary to post the events here.  Please remove these codings.

**Review by tristaZero — CHANGES_REQUESTED** [review]

**Comment by tristaZero** [comment]
@xiaoshen11 Hi thanks for this PR, please look at the failed CI tests. Besides, some comments still need your attention.

**Comment by xiaoshen11** [comment]
@tristaZero The CI still fail，I did not update shardingsphere-scaling-core module，I dont konw how to slove it.

Error:  Failed to execute goal on project shardingsphere-scaling-core: Could not resolve dependencies for project org.apache.shardingsphere:shardingsphere-scaling-core:jar:5.0.0-RC1-SNAPSHOT: Failed to collect dependencies at commons-io:commons-io:jar:2.8.0: Failed to read artifact descriptor for commons-io:commons-io:jar:2.8.0: Could not transfer artifact commons-io:commons-io:pom:2.8.0 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/commons-io/commons-io/2.8.0/commons-io-2.8.0.pom: Connection reset -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <args> -rf :shardingsphere-scaling-core
Error: Process completed with exit code 1.

**Comment by tristaZero** [comment]
Hi @xiaoshen11 Did you see my reply on the relevant issue?
Now, the CI passed but it is necessary to take care of the comments. Thanks.

**Comment by xiaoshen11** [comment]
@tristaZero i see，i hava change the code

File: shardingsphere-infra/shardingsphere-infra-optimize/src/main/java/org/apache/shardingsphere/infra/optimize/core/metadata/refresher/type/DropTableStatementFederateRefresher.java (hunk 2 above)
Comment (xiaoshen11) [reply shardingsphere-infra/shardingsphere-infra-optimize/src/main/java/org/apache/shardingsphere/infra/optimize/core/metadata/refresher/type/DropTableStatementFederateRefresher.java]: throws SQLException {
sqlStatement.getTables().forEach(each -> schema.remove(each.getTableName().getIdentifier().getValue()));

this line also need remove？

**Review by tristaZero — APPROVED** [review]
Hey @xiaoshen11 ,

Looks nice. ;-)

**Comment by xiaoshen11** [comment]
@tristaZero Thanks for your guidance

**Pull request merged**